backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/AbstractPPOrderReceiptHUProducer.java

```diff
@@ -419,6 +405,27 @@ private void updateReceivedHUs(
 		addAssignedHUs(hus);
 	}
 
+	@Nullable
+	private String getOrLoadLotNumber()
+	{
+		if (lotNumber == null)
```

Comment:
in case there is no lot numer configured, this method will be checked for each HU. isn't it?

Reply:
yes, but on the first call a lotNumber will be set. See line #423

Reply:
addressed